### ISSUE TITLE
upgrading chart.js to 3.1.1 and getting many errors

### ISSUE THREAD

**Issue by sofipr** [valor-software/ng2-charts#1309]
hey 
i'm getting this errors: 

   ERROR in node_modules/ng2-charts/lib/base-chart.directive.d.ts:4:30 - error TS2724: Module '"../../chart.js/types/index.esm"' has no exported member 'ChartDataSets'. Did you mean 'ChartDataset'?
    
    4 import { ChartConfiguration, ChartDataSets, ChartOptions, ChartPoint, ChartType, PluginServiceGlobalRegistration, PluginServiceRegistrationOptions } from 'chart.js';
                                   ~~~~~~~~~~~~~
    node_modules/ng2-charts/lib/base-chart.directive.d.ts:4:59 - error TS2305: Module '"../../chart.js/types/index.esm"' has no exported member 'ChartPoint'.
    
    4 import { ChartConfiguration, ChartDataSets, ChartOptions, ChartPoint, ChartType, PluginServiceGlobalRegistration, PluginServiceRegistrationOptions } from 'chart.js';
                                                                ~~~~~~~~~~
    node_modules/ng2-charts/lib/base-chart.directive.d.ts:4:82 - error TS2305: Module '"../../chart.js/types/index.esm"' has no exported member 'PluginServiceGlobalRegistration'.
    
    4 import { ChartConfiguration, ChartDataSets, ChartOptions, ChartPoint, ChartType, PluginServiceGlobalRegistration, PluginServiceRegistrationOptions } from 'chart.js';
                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/ng2-charts/lib/base-chart.directive.d.ts:4:115 - error TS2305: Module '"../../chart.js/types/index.esm"' has no exported member 'PluginServiceRegistrationOptions'.
    
    4 import { ChartConfiguration, ChartDataSets, ChartOptions, ChartPoint, ChartType, PluginServiceGlobalRegistration, PluginServiceRegistrationOptions } from 'chart.js';
                                                                                                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/ng2-charts/lib/base-chart.directive.d.ts:33:12 - error TS2304: Cannot find name 'Chart'.
    
    33     chart: Chart;
                  ~~~~~
    node_modules/ng2-charts/lib/base-chart.directive.d.ts:57:35 - error TS2304: Cannot find name 'Chart'.
    
    57     getChartBuilder(ctx: string): Chart;

**Comment by RoboZoom** [comment]
I am also experiencing this issue - I checked my package.json, and did a clean install.  It looks like there is bad type mapping, but I haven't figured this out yet.

**Comment by Parsa-Sedigh** [comment]
Any updates? I'm using angular v12.0.0 - chartjs v3.2.1 and ng2-charts v2.4.2

**Comment by weirdyang** [comment]
Is there a fix for this?

`node_modules/ng2-charts/lib/base-chart.directive.d.ts:33:12 - error TS2304: Cannot find name 'Chart'.`

```
  "dependencies": {
    "@angular/animations": "~11.2.5",
    "@angular/cdk": "^11.2.13",
    "@angular/common": "~11.2.5",
    "@angular/compiler": "~11.2.5",
    "@angular/core": "~11.2.5",
    "@angular/forms": "~11.2.5",
    "@angular/material": "^11.2.13",
    "@angular/platform-browser": "~11.2.5",
    "@angular/platform-browser-dynamic": "~11.2.5",
    "@angular/router": "~11.2.5",
    "chart.js": "^3.2.1",
    "ng2-charts": "^2.4.2",
    "rxjs": "~6.6.0",
    "tslib": "^2.0.0",
    "zone.js": "~0.11.3"
  },
```

**Comment by Parsa-Sedigh** [comment]
@weirdyang Your error is part of this issue which still is not resolved

**Comment by santam85** [comment]
Can you try updating to ng2-charts 3.0.0-rc.1?

**Comment by Parsa-Sedigh** [comment]
Ok I'm using "ng2-charts": "^3.0.0-rc.2" and if I import the `ChartsModule`  in my component's module, it throws this error:

Uncaught (in promise): Error: ChartsModule is already loaded. Import it in the AppModule only
Error: ChartsModule is already loaded. Import it in the AppModule only
    at new ChartsModule (ng2-charts.js:264)
    at Object.ChartsModule_Factory [as factory] (ng2-charts.js:277)
    at R3Injector.hydrate (core.js:11412)
    at R3Injector.get (core.js:11232)
    at core.js:11269
    at Set.forEach (<anonymous>)
    at R3Injector._resolveInjectorDefTypes (core.js:11269)
    at new NgModuleRef$1 (core.js:25327)
    at NgModuleFactory$1.create (core.js:25381)
    at MapSubscriber.project (router.js:3676)
    at resolvePromise (zone.js:1213)
    at resolvePromise (zone.js:1167)
    at zone.js:1279
    at ZoneDelegate.invokeTask (zone.js:406)
    at Object.onInvokeTask (core.js:28654)
    at ZoneDelegate.invokeTask (zone.js:405)
    at Zone.runTask (zone.js:178)
    at drainMicroTaskQueue (zone.js:582)


And then if I import it only in `AppModule` again it throws the exact error before and also now the bindings that I use in the template, are unknown properties.

**Comment by santam85** [comment]
What is your Angular version?

**Comment by Parsa-Sedigh** [comment]
12.0.3

**Comment by santam85** [comment]
Fixed in ng2-charts 3.0.0-rc.3

**Comment by gbhowmick1** [comment]
I have used "chart.js": "^2.9.4" and the error is resolved. so you can use this version.
